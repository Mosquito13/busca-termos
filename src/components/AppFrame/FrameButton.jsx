import classNames from 'classnames';

const FrameButton = ({ close, icon, onClick }) => {
  const classes = classNames(
    'app-frame__button',
    close && 'app-frame__button--close'
  );

  return (
    <button className={classes} onClick={onClick}>
      <div className="app-frame__button-icon">
        {icon}
      </div>
    </button>
  );
};

export default FrameButton;
