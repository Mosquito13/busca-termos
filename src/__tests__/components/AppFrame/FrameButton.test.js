import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import FrameButton from '../../../components/AppFrame/FrameButton';

describe(`${FrameButton.name}`, () => {
  const iconMock = <span className="an-icon" />;

  describe('tests', () => {
    let wrapper, onClickFn;

    beforeAll(() => {
      onClickFn = jest.fn();

      wrapper = shallow(<FrameButton onClick={onClickFn} icon={iconMock} />);
    });

    it('should render default button', () => {
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(wrapper.find('.app-frame__button')).toHaveLength(1);
      expect(wrapper.find('.app-frame__button--close')).toHaveLength(0);
      expect(wrapper.find('.an-icon')).toHaveLength(1);
    });

    it('should render close button', () => {
      wrapper.setProps({ close: true });
      expect(wrapper.isEmptyRender()).toBeFalsy();
      expect(wrapper.find('.app-frame__button')).toHaveLength(1);
      expect(wrapper.find('.app-frame__button--close')).toHaveLength(1);
      expect(wrapper.find('.an-icon')).toHaveLength(1);
    });

    afterAll(() => {
      wrapper.unmount();
    });
  });

  describe('snapshots', () => {
    it('should render default button', () => {
      const tree = renderer
        .create(<FrameButton icon={iconMock} onClick={jest.fn()} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render close button', () => {
      const tree = renderer
        .create(<FrameButton close icon={iconMock} onClick={jest.fn()} />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
