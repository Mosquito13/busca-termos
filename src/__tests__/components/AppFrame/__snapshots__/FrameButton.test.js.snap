// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrameButton snapshots should render close button 1`] = `
<button
  className="app-frame__button app-frame__button--close"
  onClick={[MockFunction]}
>
  <div
    className="app-frame__button-icon"
  >
    <span
      className="an-icon"
    />
  </div>
</button>
`;

exports[`FrameButton snapshots should render default button 1`] = `
<button
  className="app-frame__button"
  onClick={[MockFunction]}
>
  <div
    className="app-frame__button-icon"
  >
    <span
      className="an-icon"
    />
  </div>
</button>
`;
